### PR TITLE
Replace lighting_console with chip_rpc.console

### PR DESCRIPTION
**Pull request by rgoliver**
#### Problem
Currently the lighting app is the only example with an RPC console which provides a simple environment for sending commands to a chip device with RPCs enabled. There are now other examples which have RPCs integrated, creating custom consoles for every application is not scaleable, instead we need a single chip RPC console with all RPC protos included.

#### Change overview

- Move/rename lighting-app.rpc_sonsole to chip-rpc.console.
- Include all protos for locking, lighting, and wifi examples.
- Move the protos from the ipv6only and lock-app examples into common folders.
- Update Pigweed, and integrate the new console changes and UI.
- Update all the example docs to use the new console.

#### Testing
The tool was tested against the following platforms/examples to ensure RPCs and logging both worked:

- EFR lighting example
- ESP locking example
- ESP ipv6only example
- NRF lighting example
- linux lighting example (logging currently does not go to debug console, instead uses stdio)

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: examples/common/pigweed/rpc_console/py/chip_rpc/console.py [hunk 1]
```diff
@@ -94,11 +102,26 @@ def _start_ipython_terminal(client: HdlcRpcClient) -> None:
...
+
+          rpcs.chip.rpc.DeviceCommon.GetDeviceInfo()
+
+          LOG.warning('Message appears console log window.')
+    """)
+
+    embed(global_vars=local_variables,
+          local_vars=None,
+          loggers=[_DEVICE_LOG],
+          repl_startup_message=welcome_message,
```
Comment (AnthonyDiGirolamo) [inline examples/common/pigweed/rpc_console/py/chip_rpc/console.py:123]: You could add an `app_title='CHIP Console',` kwarg for some branding.

**Review by bzbarsky-apple — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

File: examples/common/pigweed/rpc_console/py/chip_rpc/console.py (hunk 1 above)
Comment (rgoliver) [reply examples/common/pigweed/rpc_console/py/chip_rpc/console.py:123]: Done, thanks for the tip!

**Comment by mspang** [comment]
@rgoliver 

clang upgrade appears to have added warnings..

**Comment by mspang** [comment]
> @rgoliver
> 
> clang upgrade appears to have added warnings..

(also, I think you really ought to do these rolls in their own PR)

**Review by msandstedt — APPROVED** [review]

**Pull request merged**